### PR TITLE
docs(core): add info on NX_TARGET_TASK_PROJECT

### PR DESCRIPTION
**Pull request by dan-cooke**
Just adding some documentation on the elusive `NX_TARGET_TASK_PROJECT` feature that has been apparently in the framework for 16 months.

I was actually attempting to add a similar feature when I discovered it by digging through the source code.

### CODE REVIEW

File: packages/nx/src/tasks-runner/run-command.ts [hunk 1]
```diff
@@ -113,6 +113,8 @@ export async function runCommand(
   extraTargetDependencies: Record<string, (TargetDependencyConfig | string)[]>,
   extraOptions: { excludeTaskDependencies: boolean; loadDotEnvFiles: boolean }
 ) {
+  console.log(projectsToRun, projectGraph);
```
Comment (AgentEnder) [inline packages/nx/src/tasks-runner/run-command.ts:116]: This needs to be removed

**Review by AgentEnder — CHANGES_REQUESTED** [review]
It seems like CircleCI picked up some weird issue. Can you rebase?

File: packages/nx/src/tasks-runner/run-command.ts (hunk 1 above)
Comment (dan-cooke) [reply packages/nx/src/tasks-runner/run-command.ts]: Removed, and rebased

**Comment by isaacplmann** [comment]
@dan-cooke , I put your commit on a PR that I opened up and the CI passed.

**Comment by AgentEnder** [comment]
Thanks @isaacplmann

**Comment by dan-cooke** [comment]
@isaacplmann thank you!

**Pull request closed without merging**